File: backend/app/schemas/pagination.py
# SPDX-License-Identifier: AGPL-3.0-or-later
from pydantic import BaseModel


class Page[T](BaseModel):
    """Generic offset-paginated response shape, shared by every list endpoint."""

    items: list[T]
    total: int
    skip: int
    limit: int

File: frontend/packages/api-client/src/models/pagination.ts
// SPDX-License-Identifier: AGPL-3.0-or-later

/** Generic offset-paginated response shape, mirroring the backend's `schemas.pagination.Page`. */
export interface Page<T> {
  items: T[];
  total: number;
  skip: number;
  limit: number;
}
